crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -595,6 +595,18 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                         .change_context(errors::ApiErrorResponse::InternalServerError)
                         .attach_printable("Could not parse the connector response")?;
 
+                    let auth_update = match (
+                        authentication_data.is_some(),
+                        payment_data.payment_attempt.authentication_type,
+                    ) {
+                        (true, Some(common_enums::AuthenticationType::NoThreeDs))
+                        | (true, None) => Some(common_enums::AuthenticationType::ThreeDs),
+                        (false, Some(common_enums::AuthenticationType::ThreeDs))
+                        | (false, None) => Some(common_enums::AuthenticationType::NoThreeDs),
```

Comment:
We can't convert `ThreeDs` to `NoThreeDs` when authentication_data is None because, in a frictionless flow, we don't get the redirection data, but it's still a ThreeDS flow